### ISSUE TITLE
not working path with "+" symbol C:\localhost\coffee++\src

### ISSUE THREAD

**Issue by Maxmaxmaximus** [floatdrop/gulp-watch#206]
not working path with "+" symbol C:\localhost\coffee++\src

**Comment by UltCombo** [comment]
Can you describe what is not working as expected?

I've tried your example and I'm getting a "glob pattern string required" error thrown from Minimatch, is this the same issue you are having?

**Comment by Maxmaxmaximus** [comment]
@UltCombo oh no, just watcher not reactin in file changes. that all)

P.S. windows 10

**Comment by UltCombo** [comment]
Oh I see. I just noticed there are two issues here:

- `watch('coffee++/src/**')` - does not fire any events;
- `watch('coffee++/src')` - triggers event but throws the aforementioned error.

**Comment by UltCombo** [comment]
Looks like both issues are caused by micromatch not handling `+` properly. I've opened an upstream issue [here](https://github.com/jonschlinkert/micromatch/issues/45).

**Comment by Maxmaxmaximus** [comment]
@UltCombo its a Glob object error?

**Comment by UltCombo** [comment]
I think we can keep this issue open. Although this is an upstream issue, it affects this package's behavior directly so it is an issue here as well. Also, keeping the issue open serves a few other purposes:

- Closing an issue usually implies that it either has been fixed or will deliberately not be fixed. This is not the case here, this issue is simply blocked by an upstream issue.

- Once the upstream issue is fixed, there *might* be another issue in this package or in another upstream package still preventing this from working correctly. I'm pretty sure this is not the case for this issue specifically, but we can't really predict the future with certainty. :wink: 

- Although fixing issues upstream is most often the right thing to do, it is also often possible to add workarounds or swap dependencies as a way to temporarily workaround upstream issues. We can use this issue for this kind of discussion as well.

- Open issues have more visibility than closed ones, so:

  - Someone who runs into the same issue and searches the open issues list is more likely to find this issue and join the conversation than opening a duplicated issue.

  - Contributors willing to help often scan the open issues list and thus may find this issue and decide to help fixing the upstream issue or suggesting workarounds.

**Comment by jonschlinkert** [comment]
I believe this is (will be) fixed by https://github.com/jonschlinkert/micromatch/tree/issue-45-plus-literal.

**Comment by UltCombo** [comment]
I just pulled your branch and it fixed the issue, nice work! :smile: 

`watch('coffee++/src/**')` works correctly now with @jonschlinkert's micromatch fix.

`watch('coffee++/src')` is still throwing an error because chokidar does some non-trivial work when passed a directory path that does not contain globbing patterns. That is, it install watchers for files inside the given directory as well. I'll check if this behavior aligns with `gulp.watch` and open another issue otherwise.

**Comment by jonschlinkert** [comment]
fixed by https://github.com/jonschlinkert/micromatch/commit/49a006811afd553a86636f685a2e9ec4397e2517. Thanks for the feedback @UltCombo!

**Comment by UltCombo** [comment]
Awesome, thanks @jonschlinkert. :smile: 

@Maxmaxmaximus Please try reinstalling your project's dependencies:

```sh
rm -rf node_modules
npm install
```

And check if everything works correctly now.
